### PR TITLE
Add NetworkMode to bake target

### LINKED ISSUE #848: buildx bake not honouring docker-compose build network mode host
Assuming I have the below, and a docker container running on my machine that exposes port `3000`

**docker-compose.yml**
```
version: '3.2'

services:
  app:
    image: "test-boolean2"
    restart: always
    user: root
    build:
      context: .
      network:
        host
      dockerfile: Dockerfile
```

**Dockerfile**
```
FROM centos:8

RUN curl http://localhost:3000
```

- `docker-compose build` succeeds
- `docker buildx bake` succeeds when using `docker-container` driver with `--driver-opt network=host`
- `docker buildx build --network=host .` succeeds when using the `default` builder/`docker` driver

However `docker buildx bake` fails when using the `default` builder as it can't reach `localhost:3000`. It feels as though perhaps `buildx bake` might not be using `network: host` specified in the `docker-compose.yml` correctly?

### PR DESCRIPTION
**Pull request by zachary-povey**
Allows specification of network mode in a bake target.

Fixes #848

Signed-off-by: Zachary Povey <zachary.povey@autotrader.co.uk>

### CODE REVIEW

**Comment by tonistiigi** [comment]
I wonder if we should leave it only for compose files for now. Using `build --network` is discouraged in buildx. Instead you are supposed to use `RUN --network` inside the dockerfile https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#network-modes-run---networknonehostdefault and allow network entitlement on build. Proposal for entitlements support in bake is in https://github.com/docker/buildx/issues/179

**Comment by zachary-povey** [comment]
@tonistiigi - happy to change it to just compose files, should I do that? 

Mu use case for this is backwards compatibility with `docker-compose` anyway, it just seemed (from a naive perspective) if something was available to `build` it should be available via a `bake` target.

**Comment by tonistiigi** [comment]
> happy to change it to just compose files, should I do that?

I think so. cc @crazy-max

**Comment by crazy-max** [comment]
> 

Yes sgtm

**Comment by zachary-povey** [comment]
Ok, thanks both :) 

@tonistiigi I've pushed a commit removing support from HCL/JSON files. Did you also want me to remove the ability to specify it via a cmd line override with `--set`?

File: bake/bake.go [hunk 1]
```diff
@@ -563,6 +567,8 @@ func (t *Target) AddOverrides(overrides map[string]Override) error {
 					}
 				}
 			}
+		case "network":
+			t.NetworkMode = &o.Value
```
Comment (crazy-max) [inline bake/bake.go:571]: > Did you also want me to remove the ability to specify it via a cmd line override with `--set`?

Yes remove the override for this case.

Comment (zachary-povey) [reply bake/bake.go]: Ok that's done now 👍

**Comment by zachary-povey** [comment]
@tonistiigi @crazy-max - think this is good to go now?

**Review by crazy-max — APPROVED** [review]
LGTM

**Pull request merged**